pkg/actions/cluster/get.go

```diff
@@ -0,0 +1,139 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	"github.com/kris-nova/logger"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+	"github.com/weaveworks/eksctl/pkg/eks"
+)
+
+const (
+	eksctlCreatedTrue    api.EKSCTLCreated = "True"
+	eksctlCreatedFalse   api.EKSCTLCreated = "False"
+	eksctlCreatedUnknown api.EKSCTLCreated = "Unknown"
+)
+
+type Description struct {
+	Name   string
+	Region string
+	Owned  api.EKSCTLCreated
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fakes/fake_aws_provider.go . ProviderConstructor
+type ProviderConstructor func(spec *api.ProviderConfig, clusterSpec *api.ClusterConfig) (*eks.ClusterProvider, error)
+
+//counterfeiter:generate -o fakes/fake_stack_provider.go . StackManagerConstructor
+type StackManagerConstructor func(provider api.ClusterProvider, spec *api.ClusterConfig) manager.StackManager
+
+var (
+	newClusterProvider ProviderConstructor     = eks.New
+	newStackCollection StackManagerConstructor = manager.NewStackCollection
+)
+
+func GetClusters(provider api.ClusterProvider, listAllRegions bool, chunkSize int) ([]Description, error) {
+	if !listAllRegions {
+		return listClusters(provider, int64(chunkSize))
+	}
+
+	var clusters []Description
+	authorizedRegionsList, err := provider.EC2().DescribeRegions(&ec2.DescribeRegionsInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe regions: %w", err)
+	}
+
+	authorizedRegions := map[string]struct{}{}
+	for _, r := range authorizedRegionsList.Regions {
+		authorizedRegions[*r.RegionName] = struct{}{}
+	}
+
+	for _, region := range api.SupportedRegions() {
+		if _, authorized := authorizedRegions[region]; !authorized {
+			continue
+		}
+		// Reset region and recreate the client.
+		ctl, err := newClusterProvider(&api.ProviderConfig{
+			Region:      region,
+			Profile:     provider.Profile(),
+			WaitTimeout: provider.WaitTimeout(),
+		}, nil)
+
+		if err != nil {
+			logger.Critical("error creating provider in %q region: %v", region, err)
+			continue
+		}
+
+		newClusters, err := listClusters(ctl.Provider, int64(chunkSize))
+		if err != nil {
```

Comment:
If I see that right, I don't see an error case for this scenario, when there is a failure and the cluster is not added.

Reply:
you caught me. I was being lazy

Reply:
understandable